src/Solver/time_stepping/TimeCluster.cpp

```diff
@@ -695,7 +695,7 @@ void TimeCluster::correct() {
   // First cluster calls fault receiver output
   // Call fault output only if both interior and copy parts of DR were computed
   // TODO: Change from iteration based to time based
-  if (m_clusterId == 0
+  if (dynamicRuptureScheduler->getIsFirstDynamicRuptureCluster()
```

Comment:
Another idea is to name the method as `isFirstNotEmptyDynamicRuptureCluster`. The name will give a hint to the others (in the future) why `m_clusterId == 0` was incorrect